jsx/FilterForm.js

```diff
@@ -18,164 +18,49 @@ import Panel from 'Panel';
-   *
-   * @return {{}} filter - filterData
-   */
-  setFilter(type, key, value) {
-    let filter = {};
-    if (this.props.filter) {
-      filter = JSON.parse(JSON.stringify(this.props.filter));
-    }
-    if (key) {
-      filter[key] = {};
-      if (value) { // all defined/non-null values must have some length, else empty string
-        filter[key].value = Object.keys(value).length > 0 ? value : '';
-      } else { // null and undefined handled here
-        filter[key].value = '';
-      }
-      filter[key].exactMatch = (type === 'SelectElement' || type === 'select');
-    }
-    if (filter && key && value === '') {
-      delete filter[key];
-    }
-    return filter;
   }
 
   /**
    * Sets filter object and querystring to reflect values of input fields
    *
-   * @param {string} type - form element type (i.e component name)
-   * @param {string} fieldName - the name of the form element
-   * @param {string} fieldValue - the value of the form element
+   * @param {string} name - form element type (i.e component name)
+   * @param {string} value - the name of the form element
+   * @param {string} id
+   * @param {string} type
    */
-  onElementUpdate(type, fieldName, fieldValue) {
-    // Make sure key is of string type and value is of string or object type
-    // before sending them to querystring
-    if (typeof fieldName !== 'string' ||
-        (typeof fieldValue !== 'string' && typeof fieldValue !== 'object')) {
-      return;
+  onElementUpdate(name, value, id, type) {
+    const filter = JSON.parse(JSON.stringify(this.props.filter));
+    const exactMatch = type === 'textbox' ? false : true;
+    if (value === null || value === '') {
+      delete filter[name];
+    } else {
+      filter[name] = {};
+      filter[name]['value'] = value;
+      filter[name]['exactMatch'] = exactMatch;
     }
 
-    // Update query string
-    let queryFieldName = (fieldName === 'candID') ? 'candidateID' : fieldName;
-    this.queryString = QueryString.set(this.queryString, queryFieldName, fieldValue);
-
-    // Update filter and get new filter object
-    let filter = this.setFilter(type, fieldName, fieldValue);
-    this.props.onUpdate(filter);
+    this.props.updateFilter(filter);
   }
 
   render() {
-    // Get formatted children
-    let formChildren = this.getFormChildren();
-    let formElements = this.props.formElements;
-
-    if (formElements) {
-      Object.keys(formElements).forEach(function(fieldName) {
-        let queryFieldName = (fieldName === 'candID') ? 'candidateID' : fieldName;
-        formElements[fieldName].onUserInput = this.onElementUpdate.bind(null, formElements[fieldName].type);
-        formElements[fieldName].value = this.queryString[queryFieldName];
-      }.bind(this));
-    }
+    const formChildren = React.Children.map(this.props.children, (child) => {
+      return React.cloneElement(
+        child,
+        {
+          ...child.props,
+          value: (this.props.filter[child.props.name]||{}).value,
```

Comment:
nitpick but could you put some spaces in `this.props.filter[child.props.name]||{}`

Reply:
For sure.